extensions/sql-database-projects/src/models/project.ts

```diff
@@ -254,24 +265,40 @@ export class Project {
 	}
 
 	private addDatabaseReferenceToProjFile(entry: DatabaseReferenceProjectEntry): void {
-		const referenceNode = this.projFileXmlDoc.createElement(constants.ArtifactReference);
-		referenceNode.setAttribute(constants.Condition, constants.NetCoreCondition);
-		referenceNode.setAttribute(constants.Include, entry.isSystemDatabase ? entry.fsUri.fsPath.substring(1) : entry.fsUri.fsPath); // need to remove the leading slash for system database path for build to work on Windows
+		let referenceNode = this.projFileXmlDoc.createElement(constants.ArtifactReference);
+
+		// if it's a system database reference, we'll add an additional node with the SSDT location of the dacpac
+		if ((<SystemDatabaseReferenceProjectEntry>entry).ssdtUri) {
+			referenceNode.setAttribute(constants.Condition, constants.NetCoreCondition);
+		}
+
+		referenceNode.setAttribute(constants.Include, (<SystemDatabaseReferenceProjectEntry>entry).ssdtUri ? entry.fsUri.fsPath.substring(1) : entry.fsUri.fsPath); // need to remove the leading slash for system database path for build to work on Windows
+		this.addDatabaseReferenceChildren(referenceNode, entry.name);
+		this.findOrCreateItemGroup(constants.ArtifactReference).appendChild(referenceNode);
+		this.databaseReferences.push(path.parse(entry.fsUri.fsPath.toString()).name);
 
+		// add a reference to the dacpac in SSDT if it's a system db
+		if ((<SystemDatabaseReferenceProjectEntry>entry).ssdtUri) {
+			let ssdtReferenceNode = this.projFileXmlDoc.createElement(constants.ArtifactReference);
+			ssdtReferenceNode.setAttribute(constants.Condition, constants.NotNetCoreCondition);
+			ssdtReferenceNode.setAttribute(constants.Include, (<SystemDatabaseReferenceProjectEntry>entry).ssdtUri.fsPath.substring(1)); // need to remove the leading slash for system database path for build to work on Windows
+			this.addDatabaseReferenceChildren(ssdtReferenceNode, entry.name);
+			this.findOrCreateItemGroup(constants.ArtifactReference).appendChild(ssdtReferenceNode);
+		}
```

Comment:
I kept them separate so that the logic for creating the two different nodes is separate. The database reference node will always be created, but the additional reference for SSDT is only added if it's a system database reference

Reply:
Sounds good. And the additional comments update are helpful as well. Thanks!